instance/models/mixins/secret_keys.py

```diff
@@ -90,7 +85,7 @@ def get_secret_key_settings(self):
         """
         key_provider = self.get_key_provider()
         if not self.secret_key_b64encoded:
-            logger.warning('Instance %s does not have a secret key; not prefilling variables.', self)
+            self.logger.warning('Instance %s does not have a secret key; not prefilling variables.', self)
```

Comment:
FYI, you don't need to add `%s..., self` here, because the instance's `logger` includes that info. It's not doing any harm though :)